src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,32 @@
 package mate.academy;
 
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Executor;
 
 public class AsyncRequestProcessor {
     private final Executor executor;
+    private Map<String, UserData> cache = new ConcurrentHashMap<>();
 
     public AsyncRequestProcessor(Executor executor) {
         this.executor = executor;
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        for (String key : cache.keySet()) {
+            if (key.equals(userId)) {
+                return CompletableFuture.supplyAsync(() -> cache.get(key));
```

Comment:
The use of `CompletableFuture.supplyAsync()` is correct here, but you should pass the `executor` to it to ensure that the task is executed asynchronously using the provided executor.